src/lib.rs

```diff
@@ -215,14 +229,14 @@ pub struct TestFile {
 }
 
 impl TestFile {
-    fn new(filename: &str) -> Result<Self, Error> {
-        let contents = fs::read_to_string(filename)
-            .with_context(|| format!("error reading file {}", filename))?;
+    fn new(filename: &PathBuf) -> Result<Self, Error> {
+        let contents = fs::read_to_string(filename.to_str().unwrap())
+            .with_context(|| format!("error reading file {}", filename.to_str().unwrap()))?;
```

Comment:
The idiom for displaying paths is `filename.display()`, which gives you back a thing that is display-able.